velox/dwio/dwrf/utils/ProtoUtils.cpp

```diff
@@ -57,9 +57,22 @@ void ProtoUtils::writeType(
   if (parent) {
     parent->add_subtypes(footer.types_size() - 1);
   }
-  auto kind =
-      VELOX_STATIC_FIELD_DYNAMIC_DISPATCH(SchemaType, kind, type.kind());
-  self->set_kind(kind);
+  bool kindSet = false;
+  if (type.kind() == TypeKind::HUGEINT) {
+    // Hugeint is not supported by DWRF, and this branch is only for ORC
```

Comment:
The decimal writer is expected to be merged soon, can we leverage that instead of relying on this hack?

Reply:
I added the decimal ORC writer https://github.com/facebookincubator/velox/commit/da6c05f20a73b641b3bb21a10d6ab42d9a2e4c10 several months ago, but the footer is still incompatible, seeing https://github.com/facebookincubator/velox/issues/12418.